driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -21,12 +21,68 @@
 import java.util.function.Function;
 
 /**
```

Comment:
[optional]

I propose us to separate the documentation of the whole API (the place for it is the package docs) and the documentation of the `Expression` class. But I am neither insisting on it nor volunteering to do the work of separating it.

Reply:
I cannot clearly identify docs presently on the Expression class that are not fairly related to it, and so do not belong there, but instead belong to the package. At the same time, there should probably be some package docs (or at least a link to the Expression class and the Expressions class), but I do not have some specific text to put there.

Reply:
The thing about package docs is that they are harder (impossible?) to see directly in the IDE, which is where I suspect most people actually look at the javadoc.

Reply:
In IntelliJ one may view package docs in exactly the same way one views other docs:

- either hover the mouse for some time over the last name section of the package name you are interested in anywhere in the code where a package name is used;
- or place the caret on the same thing and press you hotkey combination for viewing docs.